frontend/src/components/CognitoLoginAndReg.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import Button from 'react-bootstrap/Button';
+import { useLocation } from 'react-router-dom'
+
+function useQuery() {
+  return new URLSearchParams(useLocation().search);
+}
+
+export default () => {
+  const [, setTokenObj ] = useState({});
+  const [, setUserObj ] = useState({});
+
+  let query = useQuery();
+  let code = query.get("code");
+
+  function fetchTokens() {
+    const myHeaders = new Headers();
+    myHeaders.append("Content-Type", "application/x-www-form-urlencoded");
+
+    const urlencoded = new URLSearchParams();
+    urlencoded.append("grant_type", "authorization_code");
+    urlencoded.append("client_id", "133b172mr6hvdr48eg0mb2uirh");
```

Comment:
Are you sure that this ID is something that may be visible publicly? (I assume that there's some sort of secret in another place, but I would like to make sure.)

Reply:
Its visible. It would be displayed on in the browser search bar as part of the url during login